pdl-compiler/src/parser.rs

```diff
@@ -191,26 +176,32 @@ pub struct PDLParser;
 
 type Node<'i> = Pair<'i, Rule>;
 type NodeIterator<'i> = Peekable<Filter<Pairs<'i, Rule>, fn(&Node<'i>) -> bool>>;
-type Context<'a> = (crate::ast::FileId, &'a Vec<usize>);
+struct Context<'a>(ast::FileId, &'a Vec<usize>, std::cell::Cell<usize>);
```

Comment:
I saw the `.2` below and wonder if it wouldn't be better to use a struct with named fields here?